addon/-private/system/relationships/state/has-many.js

```diff
@@ -201,17 +201,25 @@ export default class ManyRelationship extends Relationship {
   }
 
   setInitialInternalModels(internalModels) {
-    if (!internalModels) {
+    if (Array.isArray(internalModels) === false || internalModels.length === 0) {
```

Comment:
Also, the above was more of a refinement on the existing code. But yes, a future cleanup PR sounds good.